### PR TITLE
Notify STS when encryption keys are updated in azurecore

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Sends notification to backend SqlToolsService to update encryption keys in a cross-platform compatible way.. 
Addresses https://github.com/microsoft/azuredatastudio/issues/22364 paired with https://github.com/microsoft/sqltoolsservice/pull/1955

### CODE REVIEW

**Review by kburtram — APPROVED** [review]

File: extensions/azurecore/src/azurecore.d.ts [hunk 1]
```diff
@@ -314,8 +314,12 @@ declare module 'azurecore' {
 		getRegionDisplayName(region?: string): string;
 		getProviderMetadataForAccount(account: AzureAccount): AzureAccountProviderMetadata;
 		provideResources(): azureResource.IAzureResourceProvider[];
-
 		runGraphQuery<T extends azureResource.AzureGraphResource>(account: AzureAccount, subscriptions: azureResource.AzureResourceSubscription[], ignoreErrors: boolean, query: string): Promise<ResourceQueryResult<T>>;
+		/**
+		 * Event emitted when encryption keys are updated in credential store.
+		 * Returns Credential Ids for connection providers use.
```
Comment (cheenamalhotra) [inline extensions/azurecore/src/azurecore.d.ts:320]: Nit: Fix doc:

```suggestion
		 * Returns encryption keys for connection providers use.
```

File: extensions/azurecore/src/extension.ts [hunk 2]
```diff
@@ -100,7 +100,8 @@ export async function activate(context: vscode.ExtensionContext): Promise<azurec
 	updatePiiLoggingLevel();
 
 	// Create the provider service and activate
-	initAzureAccountProvider(extensionContext, storagePath, authLibrary!).catch((err) => Logger.error(err));
+	let providerService = await initAzureAccountProvider(extensionContext, storagePath, authLibrary!).catch((err) => Logger.error(err));
```
Comment (alanrenmsft) [inline extensions/azurecore/src/extension.ts:103]: Fail to init the account provider is not recoverable, right? should be continue in that case?

File: extensions/azurecore/src/azurecore.d.ts [hunk 3]
```diff
@@ -314,8 +314,12 @@ declare module 'azurecore' {
 		getRegionDisplayName(region?: string): string;
 		getProviderMetadataForAccount(account: AzureAccount): AzureAccountProviderMetadata;
 		provideResources(): azureResource.IAzureResourceProvider[];
-
 		runGraphQuery<T extends azureResource.AzureGraphResource>(account: AzureAccount, subscriptions: azureResource.AzureResourceSubscription[], ignoreErrors: boolean, query: string): Promise<ResourceQueryResult<T>>;
+		/**
+		 * Event emitted when encryption keys are updated in credential store.
+		 * Returns encryption keys for connection providers use.
+		 */
+		getOnEncryptionKeysUpdated: vscode.Event<CacheEncryptionKeys>;
```
Comment (alanrenmsft) [inline extensions/azurecore/src/azurecore.d.ts:322]: rename to `onEncryptionKeysUpdated`

File: extensions/azurecore/src/azurecore.d.ts [hunk 4]
```diff
@@ -314,8 +314,12 @@ declare module 'azurecore' {
 		getRegionDisplayName(region?: string): string;
 		getProviderMetadataForAccount(account: AzureAccount): AzureAccountProviderMetadata;
 		provideResources(): azureResource.IAzureResourceProvider[];
-
 		runGraphQuery<T extends azureResource.AzureGraphResource>(account: AzureAccount, subscriptions: azureResource.AzureResourceSubscription[], ignoreErrors: boolean, query: string): Promise<ResourceQueryResult<T>>;
+		/**
+		 * Event emitted when encryption keys are updated in credential store.
+		 * Returns encryption keys for connection providers use.
```
Comment (alanrenmsft) [inline extensions/azurecore/src/azurecore.d.ts:320]: explain what the keys are used for.

File: extensions/mssql/src/sqlToolsServer.ts [hunk 5]
```diff
@@ -90,6 +92,35 @@ export class SqlToolsServer {
...
+						iv: keys.iv
+					});
+			});
+		}
+	}
+
+	private async getAzureCoreAPI(): Promise<azurecore.IExtension> {
+		const api = (await vscode.extensions.getExtension(azurecore.extension.name)?.activate()) as azurecore.IExtension;
+		if (!api) {
+			throw new Error('azure core API undefined for mssql extension');
```
Comment (alanrenmsft) [inline extensions/mssql/src/sqlToolsServer.ts:119]: azure core extension is not activated.

**Pull request merged**